### PR TITLE
Update templates for Debian 11.5 release

### PR DESCRIPTION
**Pull request by mmguero**
## Description
Update templates for Debian 11.5 release. This includes the `README.md` file, the `builds.yml` file, and the debian-specific packer templates. 

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: packer_templates/debian/debian-11.5-amd64.json [hunk 1]
```diff
@@ -141,17 +141,17 @@
     "http_directory": "{{template_dir}}/http",
     "http_proxy": "{{env `http_proxy`}}",
     "https_proxy": "{{env `https_proxy`}}",
-    "iso_checksum": "d490a35d36030592839f24e468a5b818c919943967012037d6ab3d65d030ef7f",
-    "iso_name": "debian-11.4.0-amd64-netinst.iso",
+    "iso_checksum": "e307d0e583b4a8f7e5b436f8413d4707dd4242b70aea61eb08591dc0378522f3",
```
Comment (mmguero) [inline packer_templates/debian/debian-11.5-amd64.json:144]: see http://cdimage.debian.org/cdimage/release/11.5.0/amd64/iso-cd/SHA256SUMS

File: packer_templates/debian/debian-11.5-arm64.json [hunk 2]
```diff
@@ -87,16 +87,16 @@
     "http_directory": "{{template_dir}}/http",
     "http_proxy": "{{env `http_proxy`}}",
     "https_proxy": "{{env `https_proxy`}}",
-    "iso_checksum": "0193e7b555011a6f8fa6ad132066873fad4a5396a9ebfca8fd179122bb4c7fdd",
-    "iso_name": "debian-11.4.0-arm64-netinst.iso",
+    "iso_checksum": "70b19f7f0178d594b409822c724c2b7c0834716df1ad7e13dd10e1580380c7c9",
```
Comment (mmguero) [inline packer_templates/debian/debian-11.5-arm64.json:90]: see http://cdimage.debian.org/cdimage/release/11.5.0/arm64/iso-cd/SHA256SUMS

File: packer_templates/debian/debian-11.5-i386.json [hunk 3]
```diff
@@ -208,16 +208,16 @@
     "http_directory": "{{template_dir}}/http",
     "http_proxy": "{{env `http_proxy`}}",
     "https_proxy": "{{env `https_proxy`}}",
-    "iso_checksum": "b369b7eaeee2e897649bf3a8d00036e596149605456d038cad8da4a809333f3d",
-    "iso_name": "debian-11.4.0-i386-netinst.iso",
+    "iso_checksum": "700a775805262e16858f64eebe1a3317edef568fc44a5d55fa6f8ce48301acb2",
```
Comment (mmguero) [inline packer_templates/debian/debian-11.5-i386.json:211]: see http://cdimage.debian.org/cdimage/release/11.5.0/i386/iso-cd/SHA256SUMS

**Review by Stromweld — APPROVED** [review]
lgtm

**Pull request merged**